src/__tests__/formatPkg.test.js

```diff
@@ -188,29 +188,26 @@ describe('adds TypeScript information', () => {
     ).toEqual(expect.objectContaining({ types: { ts: 'included' } }));
   });
 
-  it('adds types possible if we can find a main file', () => {
+  it('adds types if included in the dependencies', () => {
     expect(
       formatPkg({
         name: 'xxx',
+        main: 'shell-script.sh',
         lastPublisher: { name: 'unknown' },
-        main: 'main.js',
-      })
-    ).toEqual(
-      expect.objectContaining({
-        types: { ts: { possible: true, dtsMain: 'main.d.ts' } },
+        dependencies: { typescript: '^2.3.1' },
       })
-    );
+    ).toEqual(expect.objectContaining({ types: { ts: 'included' } }));
```

Comment:
you don't have to expose a `d.ts` to import typescript code

Reply:
would it work correctly though? I feel like it would work just like JS, and not like TS, unless the main file is a .ts or .d.ts, which you don't have a guarantee on if simply typescript is in a devdep

Reply:
Yeah, I'm afraid this is a lossy heuristic - transpiled typescript without d.ts files is the same as a library with no typescript support for external users. The indication of typescript in your deps is only a very solid possibility that it has types, but there's quite a few cases where people just drop in a d.ts to the root and call it a day which this would miss. 

This change makes the results a probably whereas before it was certain. If speed is the issue, perhaps this check can be looked up vs the older index? As removal of types or switching between `@types` and included is rare

Reply:
True, but until now the Typescript check is as-loosy as this one. 

We currently call `@types/` then rollback on `types` prop (good) then we try files in unpkg (loosy). I don't see how it was certainty before.

If you don't have a `types` prop, no `typescript` deps I assume Typescript will no work anyway (or is typescript loading any .d.ts in the folder anyway ?)

Reply:
One thing to improve the certainty would be to use JsDelivr flat list files (like we talked before with Haroen), that way we just would have to iterate on all files published to npm and see if any has `.d.ts` 
I wanted to avoid doing an extra call, but there seems to be no other way to do it properly

Reply:
The fs is the source of truth for whether a library supports type definitions. If those files don't resolve, then it doesn't support definitions. Using unpkg here is not loose, it's a source of truth 👍 

The check is a re-implementation of how TypeScript's resolver looks itself, ts only cares about resolving the files when looking for definitions. The `types` and `typings` option on a package give you an extra place to redirect where the types live, but don't represent whether a library supports TS that's just another check when resolving any import/require:

```sh
mkdir -p tsd/node_modules/ex
echo "{}" > tsd/node_modules/ex/package.json
echo "export const hello = 'OK'" > tsd/node_modules/ex/index.d.ts
echo "import {hello} from 'ex'" > tsd/file.ts
cd tsd
code .
```

`hello` will be classed as `"OK"` in your editor because used all the defaults to resolve the types. 

I'm not sure about the flat list approach, but that also sounds good - you do only have to resolve one file to be certain - which I think this currently does via a head request, so I'm not sure if that'd be faster or not

Reply:
Fair enough 👍 
let's restore a proper file check then